standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java

```diff
@@ -1274,7 +1308,46 @@ public void createDatabase(Database db)
     if (!db.isSetCatalogName()) {
       db.setCatalogName(getDefaultCatalog(conf));
     }
-    client.create_database(db);
+    CreateDatabaseRequest req = new CreateDatabaseRequest();
+    req.setDatabaseName(db.getName());
+    if (db.isSetDescription()) {
+      req.setDescription(db.getDescription());
+    }
+    if (db.isSetLocationUri()) {
+      req.setLocationUri(db.getLocationUri());
+    }
+    if (db.isSetParameters()) {
+      req.setParameters(db.getParameters());
+    }
+    if (db.isSetPrivileges()) {
+      req.setPrivileges(db.getPrivileges());
+    }
+    if (db.isSetOwnerName()) {
+      req.setOwnerName(db.getOwnerName());
+    }
+    if (db.isSetOwnerType()) {
```

Comment:
we can set property directly without null check, perhaps a database builder is helpful, for example:
`
new CreateDatabaseRequest.Builder(catName/**required args*/, dbName/**required args*/).createTime(time).location(uri).managedLocation(location).build()
`
This works for other request as well, such as `AppendPartitionsRequest`, `AddPartitionsRequest`, etc.

Reply:
Thrift objects don't come with Builder methods. So I don't think there is a way to implement your suggestion here.